### PR TITLE
[10.0] port stock_auto_move to 10.0

### PR DESCRIPTION
**Pull request by Cedric-Pigeon**

### CODE REVIEW

File: stock_auto_move/models/stock_auto_move.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        moves = self.filtered(lambda m: m.state == 'assigned' and m.auto_move)
+        todo_pickings = moves.mapped('picking_id')
+        # We create packing operations to keep packing if any
+        todo_pickings.do_prepare_partial()
+        moves.action_done()
+
+    @api.multi
+    def _change_procurement_group(self):
+        automatic_group = self.env.ref('stock_auto_move.automatic_group')
+        for move in self:
```
Comment (rousseldenis) [inline stock_auto_move/models/stock_auto_move.py]: @Cedric-Pigeon You could have written `self.filtered(lambda m: m.auto_move and m.group_id != automatic_group).write({'group_id': automatic_group.id})`

File: stock_auto_move/models/stock_auto_move.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        "This can be useful for situations with chained moves where we "
+        "do not want an operator action.")
+
+
+class ProcurementOrder(models.Model):
+    _inherit = 'procurement.order'
+
+    def _get_stock_move_values(self):
+        res = super(ProcurementOrder, self)._get_stock_move_values()
+        res.update({'auto_move': self.rule_id.auto_move})
```
Comment (rousseldenis) [inline stock_auto_move/models/stock_auto_move.py]: @Cedric-Pigeon Careful if rule_id is not defined

**Review by rousseldenis — CHANGES_REQUESTED** [review]

File: stock_auto_move/models/stock_auto_move.py (hunk 1 above)
Comment (lasley) [reply stock_auto_move/models/stock_auto_move.py]: Agreed with filter, but the write should be split into an independent operation:
```
moves = self.filtered(
    lambda m: m.auto_move and m.group_id != automatic_group
)
moves.write({'group_id': automatic_group.id})
```

**Comment by Cedric-Pigeon** [comment]
@lasley @rousseldenis Changes done. Thanks.

File: stock_auto_move/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2014 NDP Systèmes (<http://www.ndp-systemes.fr>).
```
Comment (lasley) [inline stock_auto_move/__manifest__.py]: Please use the [short header](https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3) instead

File: stock_auto_move/data/stock_auto_move_data.xml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
```
Comment (lasley) [inline stock_auto_move/data/stock_auto_move_data.xml]: You can remove the data tag when using the new odoo tag

File: stock_auto_move/data/stock_auto_move_data.xml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+        <record model="procurement.group" id="automatic_group">
+            <field name="name">Automatic</field>
+            <field name="move_type">direct</field>
+        </record>
+
+    </data>
+</odoo>
```
Comment (lasley) [inline stock_auto_move/data/stock_auto_move_data.xml]: Please add a new line at the end of this file.

File: stock_auto_move/demo/stock_auto_move_demo.xml [hunk 6]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
```
Comment (lasley) [inline stock_auto_move/demo/stock_auto_move_demo.xml]: Remove data tag

File: stock_auto_move/models/stock_auto_move.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+    auto_move = fields.Boolean(
+        "Automatic move",
+        help="If this option is selected, the move will be automatically "
+        "processed as soon as the products are available.")
+
+    @api.multi
+    def action_assign(self):
+        super(StockMove, self).action_assign()
+        # picking_env = self.env['stock.picking']
```
Comment (lasley) [inline stock_auto_move/models/stock_auto_move.py]: Remove commented code

File: stock_auto_move/tests/test_stock_auto_move.py [hunk 8]
```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2014 NDP Systèmes (<http://www.ndp-systemes.fr>).
```
Comment (lasley) [inline stock_auto_move/tests/test_stock_auto_move.py]: Short header

File: stock_auto_move/views/stock_auto_move_view.xml [hunk 9]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
```
Comment (lasley) [inline stock_auto_move/views/stock_auto_move_view.xml]: Remove data tag

**Review by lasley — CHANGES_REQUESTED** [review]
Code review, minor comments

**Review by rousseldenis — APPROVED** [review]
Code review

**Comment by Cedric-Pigeon** [comment]
@lasley Is that ok for you now ? thanks

**Review by lasley — APPROVED** [review]

**Comment by lasley** [comment]
Looks great, thanks @Cedric-Pigeon. Can you please squash your last three commits into one migration commit? We'll be good for merge after that 👍

**Comment by Cedric-Pigeon** [comment]
@lasley squash done. Thanks.

**Pull request merged**